tests/test_banner.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Minitest for Banner class
+#
+# Copyright (c) 2014-2018 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/banner'
+
+# TestBanner - Minitest for Banner node utility.
+class TestBanner < CiscoTestCase
+  def setup
+    # setup runs at the beginning of each test
+    super
+    no_motd
+  end
+
+  def teardown
+    # teardown runs at the end of each test
+    no_motd
+    super
+  end
+
+  def no_motd
+    # Turn the feature off for a clean test.
+    config('no banner motd')
+  end
+
+  # TESTS
+
+  def test_motd
+    id = 'default'
+
+    banner = Cisco::Banner.new(id)
+    assert_includes(Cisco::Banner.banners, id)
+    assert_equal(Cisco::Banner.banners[id], banner)
+
+    assert_nil(Cisco::Banner.banners['default'].motd)
+    assert_nil(banner.motd)
+
+    banner.motd = 'Test banner!'
+    assert_equal(Cisco::Banner.banners['default'].motd,
+                 'Test banner!')
+    assert_equal(Cisco::Banner.banners['default'].motd,
+                 banner.motd)
+
+    banner.motd = nil
+    assert_nil(Cisco::Banner.banners['default'].motd)
+    assert_nil(banner.motd)
+  end
+end
```

Comment:
It would be nice to have a multiline banner check test for versions `>=` `7.0(3)I7(4)`

Reply:
Do you have any examples handy on fencing this for that release?  Don't want to run up against unuspported platforms/versions.

Reply:
We can use the `step_unless_legacy_defect()` method.

Here is an example:

https://github.com/cisco/cisco-network-node-utils/blob/3b254e740769b700e15000196be5dd19065b2c75/tests/test_tacacs_global.rb#L80

Versions that we support and need to skip would be

```ruby
skip_versions = ['7.0.3.I[2-6]', '7.0.3.I7.[1-3]', '7.3', '8.[1-3]']
```

The fix for the n7k will likely go into `8.4`

Reply:
Don't worry about listing a defect number for now.  Just add a message indicating that multiline banners are not supported.